lib/season_stats.rb

```diff
@@ -32,11 +32,23 @@ def worst_coach
   end
 
   def most_accurate_team
-
+    teams_grouped = @game_teams.group_by(&:team_id)
+    teams_ave_accuracy = {}
+    teams_grouped.each do |team, values|
+      teams_ave_accuracy[team] = average_accuracy(values)
+    end
+    highest_accuracy = teams_ave_accuracy.max_by{|_, value| value}
+    convert_to_team_name(highest_accuracy[0])
   end
 
   def least_accurate_team
-
+    teams_grouped = @game_teams.group_by(&:team_id)
```

Comment:
Needs to be filtered by season, which will be a method I don't have the brainpower for tonight. Not ready for merging

Reply:
Got it. Thanks for getting this one started!

My rough thinking was to first filter down the Games array to only the given season (based on the string argument passed in) and then use the game_id field to select only the games from the Game_Teams array that match with the game_ids remaining in the Games array after filtering by season. 

From there, I think we could apply the rest of your method. I can try to work through what this looks like in code this afternoon, too!

Reply:
+ now that my brain is warmed up: maybe we make a Helper module method that filters Game_Teams by the Season string argument, and then we can use it in every Season Stats method.